toolchain/lexer/string_literal.cpp

```diff
@@ -377,10 +382,13 @@ static auto ExpandEscapeSequencesAndRemoveIndent(
       }
 
       if (contents.consume_front("\n")) {
-        // Trailing whitespace before a newline doesn't contribute to the string
-        // literal value.
+        // Trailing whitespace in the source before a newline doesn't contribute
+        // to the string literal value. However, escaped whitespace (like `\t`)
+        // and any whitespace just before that does contribute.
         while (!result.empty() && result.back() != '\n' &&
-               IsSpace(result.back())) {
+               IsSpace(result.back()) &&
+               (last_escape_length == 0 ||
+                result.length() > last_escape_length + 1)) {
```

Comment:
I think there's an off-by-one error here. For example, given `"""\n\\t \n"""` I think we'll have a `last_escape_length` of `1` after processing the `\t`, and `result.length()` of 2 after skipping past the `' '`, so we won't trim the trailing `' '`. If I'm right, removing the `+ 1` should fix this, and the check for `last_escape_length == 0` then becomes redundant too.
```suggestion
               result.length() > last_escape_length) {
```

Reply:
Thanks for catching this! You are completely correct. I've incorporated your suggestion & added a test for your example as well. 🙂